sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java

```diff
@@ -275,6 +300,29 @@ public void onNewUser(String authToken) {
         super.onStarted(context, channel, appSecret, transmissionTargetToken, startedFromApp);
     }
 
+    /**
+     * Called whenever the network state is updated.
+     *
+     * @param connected true if connected, false otherwise.
+     */
+    @Override
+    public void onNetworkStateUpdated(boolean connected) {
+
+        /* If device comes back online. */
+        if (connected) {
+            for (PendingOperation po : mLocalDocumentStorage.getPendingOperations()) {
+                if (PENDING_OPERATION_CREATE_VALUE.equals(po.getOperation()) ||
+                        PENDING_OPERATION_REPLACE_VALUE.equals(po.getOperation())) {
+                    instanceCreateOrUpdate(po);
+                } else if (PENDING_OPERATION_DELETE_VALUE.equals(po.getOperation())) {
+                    instanceDelete(po);
+                } else {
+                    AppCenterLog.debug(LOG_TAG, String.format("Pending operation '%s' is not supported", po.getOperation()));
```

Comment:
Because `null` is a valid scenario for pending_operation (which means there's no dirty cache), I find it confusing to log `"Pending operation NULL is not supported"` when we know that is not the case. You can filter out `null` before logging.

Reply:
This code will not be called on operations that have `null` as that value. It's only for actual pending operations. (`null` value means the operation is not pending.)